src/converters/toRpcDuration.ts

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License.
+
+import { RpcDuration } from '@azure/functions-core';
+import { Duration } from '../../types';
+import { AzFuncSystemError } from '../errors';
+import { isDefined } from '../utils/nonNull';
+
+export function toRpcDuration(dateTime: Duration | number | undefined, propertyName: string): RpcDuration | undefined {
+    if (isDefined(dateTime)) {
+        try {
+            let timeInMilliseconds: number | undefined;
+            if (typeof dateTime === 'object') {
+                const minutes = (dateTime.minutes || 0) + (dateTime.hours || 0) * 60;
+                const seconds = (dateTime.seconds || 0) + minutes * 60;
+                timeInMilliseconds = (dateTime.milliseconds || 0) + seconds * 1000;
+            } else if (typeof dateTime === 'number') {
+                timeInMilliseconds = dateTime;
+            }
+
+            if (isDefined(timeInMilliseconds) && timeInMilliseconds >= 0) {
+                return {
+                    seconds: Math.round(timeInMilliseconds / 1000),
```

Comment:
Can't we use the `nanos` instead of `seconds` here, in order to not lose granularity?

Reply:
What happens if I specify a value of less than 1 second?

Reply:
That's fair, but if we're only supporting granularity of 1 second, why are we allowing the customer to set the duration value in milliseconds not seconds? Since here we're allowing customers to specify a duration in milliseconds, it might hint that users can specify 400 ms for example. I'd prefer if either we let customers specify the numbre of seconds not milliseconds, or we throw an error or log a warning if a value would be rounded down to zero, for example if specifying 400 ms. What would even happen in that case?

Reply:
Discussed offline - should be good enough for GA as-is. We can always improve in the future